#!/usr/bin/env python
PACKAGE = "apf_pathplanner_tutorial"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("potential_field_rows",                 int_t, 0, "A int parameter", 240, 1, 500)
gen.add("potential_field_cols",                 int_t, 0, "A int parameter", 240, 1, 500)
gen.add("potential_field_resolution",           double_t, 0, "A double parameter", 0.05, 0.01, 1)
gen.add("weight_attraction_field",              double_t, 0, "A double parameter", 0.1, 0, 10)
gen.add("weight_repulsion_field",               double_t, 0, "A double parameter", 0.1, 0, 10)
gen.add("distance_threshold_repulsion_field",   double_t, 0, "A double parameter", 0.3, 0, 2)
gen.add("max_path_length",                      double_t, 0, "A double parameter", 6.0, 0.1, 10)
gen.add("path_search_range",                    int_t, 0,   "A int parameter", 1, 1, 10)

exit(gen.generate(PACKAGE, "apf_pathplanner_tutorial", "apf_pathplanner_tutorial"))